org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/GetRefactorEditHandler.java

```diff
@@ -111,6 +122,26 @@ private static PositionInformation getFirstTrackedNodePosition(LinkedProposalPos
 		return positions[0];
 	}
 
+	private static PositionInformation getFirstTrackedNodePositionBySequenceRank(LinkedProposalPositionGroupCore positionGroup) {
```

Comment:
@fbricon @testforstephen 

I think we should leverage the sequence rank information to get the first position in the position group.

If that is also the case in other proposals, we should avoid to get the first position by the array index since it's not reliable if any code refactor happens in the upstream side (jdgt.core.manipulation)

Reply:
@jdneo ok so can you fix it in this PR?

Reply:
@fbricon How about addressing this in another PR?

Since I need some time to check if the `sequence rank` is also workable in the following proposals:
- extract varaible
- extract variable (all occurrence)
- extract constant
- extract method
- extract field
- invert variable

I can file another issue to track for this.

Reply:
@jdneo fine, you can open a separate issue

Reply:
It's here: https://github.com/eclipse/eclipse.jdt.ls/issues/1180